ote/ote/tasks/classification/task.py

```diff
@@ -0,0 +1,208 @@
+"""
+ Copyright (c) 2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+
+import io
+import os
+from subprocess import run, DEVNULL, CalledProcessError
+import tempfile
+
+import numpy as np
+import onnx
+from PIL import Image
+import torch
+from torch.onnx.symbolic_helper import parse_args
+from torch.onnx.symbolic_registry import register_op
+
+from ote.interfaces.parameters import BaseTaskParameters
+from ote.interfaces.task import ITask
+from ote.interfaces.dataset import IDataset
+
+import torchreid
+from torchreid import metrics
+from torchreid.data.transforms import build_transforms
+from torchreid.data.transforms import build_inference_transform
+from torchreid.data.datasets.image import ExternalDatasetWrapper
+from torchreid.models.common import ModelInterface
+from torchreid.engine import build_engine
+from torchreid.utils import load_pretrained_weights, set_random_seed
+from scripts.default_config import (engine_run_kwargs, get_default_config,
```

Comment:
where is it?

Reply:
It's supposed to be accessible via virtualenv. Now tasks are located in OTE, but they are imported only in runtime. If the current environment provides these dependencies, task will be loaded successfully, otherwise it won't.